src/apps/yield-protocol/arbitrum/yield-protocol.lend.token-fetcher.ts

```diff
@@ -0,0 +1,180 @@
+  id: string;
+  fyToken: {
+    id: string;
+  };
+};
+
+type YieldSeriesDetails = {
+  matured: boolean;
+  baseAsset: {
+    id: string;
+  };
+  fyToken: {
+    id: string;
+    symbol: string;
+    decimals: number;
+    totalSupply: string;
+    maturity: number;
+  };
+};
+
+type YieldRes = {
+  pools: YieldPoolDetails[];
+  seriesEntities: YieldSeriesDetails[];
+};
+
+type FyTokenDataProps = {
+  matured: boolean;
+};
+
+const query = gql`
+  {
+    seriesEntities {
+      matured
+      baseAsset {
+        id
+      }
+      fyToken {
+        id
+        symbol
+        decimals
+        totalSupply
+        maturity
+      }
+    }
+    pools {
+      id
+      fyToken {
+        id
+      }
+    }
+  }
+`;
+
+@Register.TokenPositionFetcher({ appId, groupId, network })
+export class ArbitrumYieldProtocolLendTokenFetcher implements PositionFetcher<AppTokenPosition> {
+  constructor(
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(YieldProtocolContractFactory) private readonly yieldProtocolContractFactory: YieldProtocolContractFactory,
+  ) {}
+
+  private async getYieldDetails() {
+    return await this.appToolkit.helpers.theGraphHelper.request<YieldRes>({
+      endpoint: yieldV2ArbitrumSubgraph,
+      query,
+    });
+  }
+
+  private async sellFYTokenPreview(matured: boolean, poolAddress: string, baseSymbol: string, decimals: number) {
+    const multicall = this.appToolkit.getMulticall(network);
+    if (!matured) {
+      const poolContract = this.yieldProtocolContractFactory.pool({ address: poolAddress, network });
+
+      // estimated amount of base you would get for an abritrarily small fyToken value
+      try {
+        const estimate =
+          baseSymbol === 'WETH'
+            ? // use smaller unit for weth
+              (await multicall.wrap(poolContract).sellFYTokenPreview(ethers.utils.parseUnits('.01', decimals))).mul(100)
+            : await multicall.wrap(poolContract).sellFYTokenPreview(ethers.utils.parseUnits('1', decimals));
+
+        return +ethers.utils.formatUnits(estimate, decimals);
+      } catch (error) {
+        return 0;
+      }
+    }
+
+    return 1;
+  }
+
+  async getPositions() {
+    const { pools, seriesEntities } = await this.getYieldDetails();
+
+    const tokens = await Promise.all(
+      seriesEntities.map(async series => {
+        const {
+          matured,
+          baseAsset: { id: baseAddress },
+          fyToken: { id: fyTokenaddress, symbol, totalSupply: supply, decimals, maturity },
+        } = series;
+
+        // get the corresponding pool of the series
+        const pool = pools.find(p => p.fyToken.id === fyTokenaddress);
+
+        // get the corresponding base of the series
+        const baseTokens = await this.appToolkit.getBaseTokenPrices(network);
+        const underlyingToken = baseTokens.find(v => v.address === baseAddress);
+
+        if (!underlyingToken) return null;
+
+        let pricePerShare = 0;
```

Comment:
`pricePerShare` is the ratio between the price of the token and the price of the underlying token. In your case here, it should be `1` since they are redeemable 1:1.

We should document this better, and we will!

Reply:
ah that makes total sense will update!

Reply:
this has been updated